### PR TITLE
Add support for saving initialized s-rep including fix connection of fold curve

### PR DESCRIPTION
**Pull request by jcfr**

### CODE REVIEW

File: SkeletalRepresentationInitializer/Logic/vtkSlicerSkeletalRepresentationInitializerLogic.cxx [hunk 1]
```diff
@@ -1880,5 +1936,118 @@ void vtkSlicerSkeletalRepresentationInitializerLogic::GetNeighborCells(vtkPolyDa
             output->InsertNextCell(newEdge);
         }
     }
+}
+
+void vtkSlicerSkeletalRepresentationInitializerLogic::CompletePolyData(vtkPolyData *poly, vtkPolyData *output, bool isCrest)
+{
+    poly->GetLines()->InitTraversal();
+    vtkSmartPointer<vtkIdList> idList = vtkSmartPointer<vtkIdList>::New();
```
Comment (jcfr) [inline SkeletalRepresentationInitializer/Logic/vtkSlicerSkeletalRepresentationInitializerLogic.cxx]: Use `vtkNew<vtkIdList> idList;`

File: SkeletalRepresentationInitializer/Logic/vtkSlicerSkeletalRepresentationInitializerLogic.cxx [hunk 2]
```diff
@@ -1560,6 +1560,11 @@ void vtkSlicerSkeletalRepresentationInitializerLogic::BackwardFlow(int totalNum)
     DisplayResultSrep();
 }
 
+void vtkSlicerSkeletalRepresentationInitializerLogic::SetOutputPath(const std::string &outputPath)
+{
+    mOutputPath = outputPath;
+}
+
 int vtkSlicerSkeletalRepresentationInitializerLogic::ApplyTps(int totalNum)
```
Comment (jcfr) [inline SkeletalRepresentationInitializer/Logic/vtkSlicerSkeletalRepresentationInitializerLogic.cxx]: The function `ApplyTps` does seem to be usued anywhere. Should it be removed ?

Comment (jcfr) [reply SkeletalRepresentationInitializer/Logic/vtkSlicerSkeletalRepresentationInitializerLogic.cxx]: Similarly, `CalculateSpokeDirection` does not seems to be used anywhere

File: SkeletalRepresentationInitializer/Logic/vtkSlicerSkeletalRepresentationInitializerLogic.cxx [hunk 3]
```diff
@@ -1733,6 +1738,57 @@ void vtkSlicerSkeletalRepresentationInitializerLogic::DisplayResultSrep()
...
+    // add spokeLength and spokeDirection to polyData
+    vtkSmartPointer<vtkPolyData> vtpUpSpoke = vtkSmartPointer<vtkPolyData>::New();
+    vtkSmartPointer<vtkPolyData> vtpDownSpoke = vtkSmartPointer<vtkPolyData>::New();
+    vtkSmartPointer<vtkPolyData> vtpCrestSpoke = vtkSmartPointer<vtkPolyData>::New();
+    CompletePolyData(upSpoke_poly, vtpUpSpoke);
+    CompletePolyData(downSpoke_poly, vtpDownSpoke);
+    CompletePolyData(crestSpoke_poly, vtpCrestSpoke, true);
+    // save to files
+    std::string outputUpFileName, outputDownFileName, outputCrestFileName;
+    outputUpFileName = mOutputPath + "/up.vtp";
```
Comment (jcfr) [inline SkeletalRepresentationInitializer/Logic/vtkSlicerSkeletalRepresentationInitializerLogic.cxx]: Change to 

```
std::string outputUpFileName = mOutputPath + "/up.vtp";
std::string outputDownFileName = mOutputPath + "/down.vtp";
std::string outputCrestFileName = mOutputPath + "/crest.vtp";
```

File: SkeletalRepresentationInitializer/Logic/vtkSlicerSkeletalRepresentationInitializerLogic.cxx [hunk 4]
```diff
@@ -1733,6 +1738,57 @@ void vtkSlicerSkeletalRepresentationInitializerLogic::DisplayResultSrep()
...
+    output<<"  <upSpoke>"<< outputUpFileName<<"</upSpoke>"<<std::endl;
+    output<<"  <downSpoke>"<< outputDownFileName << "</downSpoke>"<<std::endl;
+    output<<"  <crestSpoke>"<< outputCrestFileName << "</crestSpoke>"<<std::endl;
+    output<<"</s-rep>"<<std::endl;
+
+    std::string header_file = mOutputPath + "/header.xml"; ;
+    std::ofstream out_file;
+    out_file.open(header_file);
+    out_file << output.rdbuf();
+    out_file.close();
```
Comment (jcfr) [inline SkeletalRepresentationInitializer/Logic/vtkSlicerSkeletalRepresentationInitializerLogic.cxx]: I am wondering if at some point a dedicated MRML node (and associated storage node), reader and writer should be added for s-rep.

Comment (jcfr) [reply SkeletalRepresentationInitializer/Logic/vtkSlicerSkeletalRepresentationInitializerLogic.cxx]: Is the format documented somewhere ?

File: SkeletalRepresentationRefiner/Logic/vtkSlicerSkeletalRepresentationRefinerLogic.cxx [hunk 5]
```diff
@@ -101,7 +108,7 @@ void vtkSlicerSkeletalRepresentationRefinerLogic::SetOutputPath(const string &ou
     mOutputPath = outputPath;
 }
 
-int iterNum = 0;
+static int iterNum = 0;
```
Comment (jcfr) [inline SkeletalRepresentationRefiner/Logic/vtkSlicerSkeletalRepresentationRefinerLogic.cxx]: Remove this variable, it is not used.

File: SkeletalRepresentationInitializer/Logic/vtkSlicerSkeletalRepresentationInitializerLogic.cxx (hunk 4 above)
Comment (ZhiyLiu) [reply SkeletalRepresentationInitializer/Logic/vtkSlicerSkeletalRepresentationInitializerLogic.cxx]: Sorry for the late reply. The format of the header file is documented in the user manual.

Comment (jcfr) [reply SkeletalRepresentationInitializer/Logic/vtkSlicerSkeletalRepresentationInitializerLogic.cxx]: I see. Then, I suggest to add a reference to the manual from the source code. 

May be here: https://github.com/KitwareMedical/SlicerSkeletalRepresentation/blob/5e396efc4abca6753ecadd5a2a71ba8ac519430e/SkeletalRepresentationInitializer/Logic/vtkSlicerSkeletalRepresentationInitializerLogic.cxx#L1625-L1628

**Pull request merged**